src/lobe.TestApp/Program.cs

```diff
@@ -11,33 +16,82 @@ class Program
     {
         static void Main(string[] args)
         {
-            var signatureFilePath = args[0];
-            var imageToClassify = args[1];
+            var parser = CreateParser();
 
-            ImageClassifier.Register("onnx", () => new OnnxImageClassifier());
+            parser.Invoke(args);
+          
+        }
 
-            using var classifier = ImageClassifier.CreateFromSignatureFile(
-                new FileInfo(signatureFilePath));
+        private static Parser CreateParser()
+        {
+            var rootCommand = new RootCommand();
 
-            if (File.Exists(imageToClassify))
-            {
-                var results = classifier.Classify(Image
-                    .Load(imageToClassify).CloneAs<Rgb24>());
+            var signatureFileOption = new Option<FileInfo>("--signature-file");
 
-                Console.WriteLine(results.Prediction.Label);
-            }
-            else if (Directory.Exists(imageToClassify))
-            {
-                var files = Directory.GetFiles(imageToClassify);
+            var imageFileOption = new Option<FileInfo>("--image-file");
+
+            var imageFolderOption = new Option<FileInfo>("--image-folder");
 
-                foreach (var file in files)
+            var predictionEndpointOption = new Option<Uri>("--prediction-endpoint");
+
+            rootCommand.AddOption(signatureFileOption);
+            rootCommand.AddOption(imageFileOption);
+            rootCommand.AddOption(imageFolderOption);
+            rootCommand.AddOption(predictionEndpointOption);
+
+            rootCommand.Handler = CommandHandler.Create<FileInfo, FileInfo, DirectoryInfo, Uri>(
+                (signatureFile, imageFile, imageFolder, predictionEndpoint) =>
```

Comment:
will this work if one of the args is missing?

Reply:
I should add check that endpoint of signature must be provided and same that either an i mange or a folder (and file system entries must exist)